### ISSUE TITLE
Entity cannot be changed by EntitySpawnEvent

### ISSUE THREAD

**Issue by Nerahikada** [pmmp/PocketMine-MP#4973]
### Issue description
For example, if someone tried to create a plugin to change the time a dropped item despawns, this code would not work.
```php
public function onEntitySpawn(EntitySpawnEvent $event) : void{
	$entity = $event->getEntity();
	if($entity instanceof ItemEntity){
		$entity->setDespawnDelay(20);
	}
}
```

### OS and versions
* PocketMine-MP: 4.2.9

**Comment by Nerahikada** [comment]
dktapps's comment in discord:
> the problem is that EntitySpawnEvent is called from inside the constructor
> so any modifications after doing `new SomeEntity` will overwrite things done by EntitySpawnEvent

**Comment by dktapps** [comment]
I'm not really sure what can be done about this in a 4.x version.

The real problem is that the constructor of the entity directly adds the entity to the world, so it's immediately "spawned".

In a future major I intend to remove this behaviour, instead requiring that the developer call `World->addEntity()` themselves. This would permit moving the call of `EntitySpawnEvent` to `addEntity()`, which would solve the problem as long as all modifications via setters were done prior to adding the entity to the world. However, this is a BC break, and isn't suitable for 4.x because of this.

**Comment by dktapps** [comment]
While working on PM5 refactors, a possible solution occurred to me:

Instead of firing this event during the constructor (which is problematic anyway), we could instead fire it at the first call of `onUpdate()` or `entityBaseTick()` if `Entity->justCreated` is `true`. This would solve the problem, but I'm unsure if it may have any unintended side effects.

Anyone have any comments on this?

**Comment by dktapps** [comment]
This solution would also be backwards-compatible, although it would have to be done in 4.10 due to behavioural change.

**Comment by dktapps** [comment]
Fixed by #5314.
